### ISSUE TITLE
Compilation problem

### ISSUE THREAD

**Issue by smarco** [sdu-hpcl/BGSA#2]
Hi, 

I'm having a problem compiling BGSA for AVX2:

```
BGSA/original/BGSA_AVX2$ make cc=gcc
gcc -march=native -w -fopenmp  -O3 global.c file.c thread.c align_core.c cal_avx.c main.c -o aligner -lpthread 
align_core.c: In function ‘align_avx’:
align_core.c:102:24: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘-’ token
     avx_data_t dvnot3to-1ormatch;
                        ^
align_core.c:102:25: error: invalid suffix "ormatch" on integer constant
     avx_data_t dvnot3to-1ormatch;
                         ^~~~~~~~
align_core.c:221:17: error: ‘dvnot3to’ undeclared (first use in this function)
                 dvnot3to-1ormatch = _mm256_or_si256(dvpos3shiftormatch, dvpos2shift);
                 ^~~~~~~~
align_core.c:221:17: note: each undeclared identifier is reported only once for each function it appears in
align_core.c:221:26: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvpos3shiftormatch, dvpos2shift);
                          ^~~~~~~~
align_core.c:222:26: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvnot3to-1ormatch, dvpos1shift);
                          ^~~~~~~~
align_core.c:222:62: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvnot3to-1ormatch, dvpos1shift);
                                                              ^~~~~~~~
align_core.c:223:26: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvnot3to-1ormatch, dvzeroshift);
                          ^~~~~~~~
align_core.c:223:62: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvnot3to-1ormatch, dvzeroshift);
                                                              ^~~~~~~~
align_core.c:224:26: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvnot3to-1ormatch, dvneg1shift);
                          ^~~~~~~~
align_core.c:224:62: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_or_si256(dvnot3to-1ormatch, dvneg1shift);
                                                              ^~~~~~~~
align_core.c:225:26: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_xor_si256(dvnot3to-1ormatch, all_ones);
                          ^~~~~~~~
align_core.c:225:63: error: invalid suffix "ormatch" on integer constant
                 dvnot3to-1ormatch = _mm256_xor_si256(dvnot3to-1ormatch, all_ones);
```

Do you know what it can be?

Cheers,

**Comment by zhangjikai** [comment]
The above problem is because the parameter name contains '-', what parameters do you use when generating the align_core.c file?

**Comment by smarco** [comment]
Ok, see if you can reproduce the following (gcc version 8.3.0 ):

```
> git clone https://github.com/sdu-hpcl/BGSA.git
> cd BGSA/generator/
> java -jar generator.jar -M 1 -I -4 -G -2 -a avx2
> mv generated/align_core.c ../original/BGSA_AVX2/
> cd ../original/BGSA_AVX2/
> make cc=gcc
gcc -march=native -w -fopenmp  -O3 global.c file.c thread.c align_core.c cal_avx.c main.c -o aligner -lpthread 
align_core.c: In function ‘align_avx’:
align_core.c:102:24: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘-’ token
     avx_data_t dvnot3to-1ormatch;
                        ^
align_core.c:102:25: error: invalid suffix "ormatch" on integer constant
     avx_data_t dvnot3to-1ormatch;
                         ^~~~~~~~
```

**Comment by ZekunYin** [comment]
I believe the reason is the variable ”dvnot3to-1ormatch“ doesn't conform to C naming convention.
We will try to fix it soon.
Best,
Zekun

**Comment by ZekunYin** [comment]
A quick fix is to replace "dvnot3to-1ormatch" by ”dvnot3to_1ormatch“.
By the way, there is a typo in line 329. It should be `sumbit8 = _mm256_and_si256(sumbit8, sumbit8);`
We will try to fix the issues and release a new version.

**Comment by zhangjikai** [comment]
Fix in #3, try with [this file](https://github.com/sdu-hpcl/BGSA/blob/master/generator/generator.jar).

**Comment by ZekunYin** [comment]
This issue will be temporarily closed.  If there is still a problem, please feel free to re-open it.
